### ISSUE TITLE
Where is the ontology?

### ISSUE THREAD

**Issue by hoijui** [1EdTech/openbadges-specification#551]
Where is the actual, machine-readable RDF ontology for Open Badge?
Be it Turtle, JSON-LD or any other of the supported formats.
There are contexts, for example:

- <https://openbadgespec.org/v2/context.json>
- <https://purl.imsglobal.org/spec/ob/v3p0/context-3.0.2.json>

But I can not find the actual ontology anywhere.

For example, the v2 context states:

- `obi: "https://w3id.org/openbadges#"`
- `Assertion: "obi:Assertion"`

So I expect <https://w3id.org/openbadges#Assertion> to give me the RDF that specifies an Open Badge Assertion.
This is the whole deal of LinkedData.

Try scanning `"https://w3id.org/openbadges"` on <https://w3id.org/openbadges>:
> "Oops! Something went wrong."

I am aware that others, even the big ones like schema.org, are not doing this either, (which is a huge problem), but there it is at least possible to find the RDF ontology somewhere with a web-search.

**Comment by xaviaracil** [comment]
Hi @hoijui,

I'm working on adding the RDF ontology. My idea is to add it via RDFa in the resulting HTML page (i.e: https://purl.imsglobal.org/spec/vc/ob/vocab.html for OB 3.0), so both humans and machines can understand the schema. I'm following the same approach as Verifiable Credentials (Context at https://www.w3.org/2018/credentials/v1 where terms resolve to `https://www.w3.org/2018/credentials#...`)

Will this work for you? I'm not such an expert in JSON-LD and RDF, so any advise will be appreciated

**Comment by hoijui** [comment]
heyy @xaviaracil ! :-)
that makes me happy!
Indeed, for humans and machines is the best. If you have it in RDFa in a git repo, it should be easy to auto-convert it to at least JSON-LD and Turtle with a CLI based tool. I would recommend at least these two formats, because .. well JSON-LD as it is the format of Open Badge data, and Turtle because it is the most compact and human-readable of the RDF formats.
Using a tool to auto-convert is also great, because if you mess up something in your HTML/RDFa, the conversion will fail, so you get a fail-fast.

I recently found this resource that explains [how to publish an ontology](https://more.metadatacenter.org/recommended-iri-patterns-ontologies-and-their-terms), and I try to apply it for our ontologies.

I am using RDF a lot since a few years, but I do not feel like an expert. a Lot of the RDF world comes to existence through academic work, and there thus there is a lot of duplication, complexity, depth, volatility and .. simply stuff that can not be used in the practice of handling real world data, only in theory/an academic context. The most expert RDF people I know, decided to create a cleaner, more practical, and almost/partly compatible alternative (<https://atomicdata.dev>). I would use it myself, if it had a way to convert from and to RDF, which it does not yet have, but it is entirely possible to do. (NOTE: I am not affiliated, and it is an open source/funded project anyway).

Feel free to ask for help in testing or review or anything like that, if you get there!

**Comment by xaviaracil** [comment]
I've updated some files to reflect that ontology:

- https://github.com/1EdTech/openbadges-specification/blob/551-where-is-the-ontology/ob_v3p0/vocab.html with the vocabulary for humans - in html - and machines - in rdfa.
- https://github.com/1EdTech/openbadges-specification/blob/551-where-is-the-ontology/ob_v3p0/vocabulary.ttl with the ontology in turtle. This file is referred from the previous by a `<a rel="alternate"...` reference.

Both files have to be validated for closing the issue. I've checked that the syntaxt is correct. Could any of you point me to a tool for check the resulted ontology?

Thanks
